src/js/providers/html5.js

```diff
@@ -159,6 +159,11 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
         webkitendfullscreen(e) {
             _fullscreenState = false;
             _sendFullscreen(e);
+        },
+
+        error() {
+            _clearVideotagSource();
```

Comment:
If you call `_clearVideotagSource` before `VideoEvents.error` you will break this error check:
```js
        } else if (errorCode === 3 || errorCode === 4) {
            code += errorCode - 1;
            if (errorCode === 4 && video.src === location.href) {
                code = HTML5_SRC_RESET;
            }
```

We should remove `error` from the mixin, since it fires HTML5 specific codes and we don't want those fired in hls or shaka.

In this work in progress, I'm overwriting the listener https://github.com/jwplayer/jwplayer-commercial/compare/master...feature/provider-error-handling#diff-9eff7f6edcda38b4ffe8c1b9c5559955R380